src/k8s/pkg/k8sd/features/cilium/network_test.go

```diff
@@ -0,0 +1,179 @@
+package cilium_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/canonical/k8s/pkg/client/helm"
+	helmmock "github.com/canonical/k8s/pkg/client/helm/mock"
+	"github.com/canonical/k8s/pkg/k8sd/features/cilium"
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/k8s/pkg/snap"
+	snapmock "github.com/canonical/k8s/pkg/snap/mock"
+	"github.com/canonical/k8s/pkg/utils"
+	"k8s.io/utils/ptr"
+)
+
+// NOTE(hue): status.Message is not checked sometimes to avoid unnecessary complexity
+
+func TestDisabled(t *testing.T) {
+	t.Run("HelmApplyFails", func(t *testing.T) {
+		applyErr := errors.New("failed to apply")
+		helmM := &helmmock.Mock{
+			ApplyErr: applyErr,
+		}
+		snapM := &snapmock.Snap{
+			Mock: snapmock.Mock{
+				HelmClient: helmM,
+			},
+		}
+		cfg := types.Network{
+			Enabled: ptr.To(false),
+		}
+
+		status, err := cilium.ApplyNetwork(context.Background(), snapM, cfg, nil)
+
+		g := NewWithT(t)
+		g.Expect(err).To(MatchError(applyErr))
```

Comment:
What's the purpose of matching the error type here? Couldn't this just be ToNot(HaveOccurred()) ?

Reply:
well I guess other errors might have happened. I think it's better for unit tests to cover specific parts/scenarios rather than 0/1 outcomes. in the end it's more of a personal preference I guess.

Reply:
In the other test cases you only use To(HaveOccurred()), so this one needs to change or the others need to change for consistency

Reply:
right, changing this needs us to either return a known error from `cilium.ApplyNetwork` or `utils.ParseCIDR`. I think it's better to be accurate were possible and just leave it be were it needs unnecessary complexity.